includes/classes/ZenShipping.php

```diff
@@ -71,7 +71,8 @@ abstract public function install(): void;
      * Remove the module's settings
      *
      */
-    public function remove() {
+    public function remove(): void
+    {
         global $db;
         $db->Execute("delete from " . TABLE_CONFIGURATION . " where configuration_key like '" . $this->class_key . "_%'");
```

Comment:
One more thing, I'm _pretty sure_ that the `_` needs to be escaped: `"\_%'.  I ran into this when recently.

Reply:
can you provide a mysql mode setting where this is required?  i would like to test it to see how it would fail and then fix accordingly.

`$this->class_key` would probably need to get escaped as well.  and IMO, i would think this should get done in the `db` class. 

in looking at the code, it seems that `bind_vars` escapes as necessary.  

would this be more appropriate:

```php
global $db;
        $sql = "DELETE FROM " . TABLE_CONFIGURATION . " WHERE configuration_key LIKE :key";
        $sql = $db->bindVars($sql, ':key', $this->class_key . '_%', 'string');
        $db->Execute($sql);
```

Reply:
I'd run across this when developing the `spam_cleanup_check` for v2.0.0.  When searching for `%_SRC_%`, I was getting _false positives_ for an email address like `########@src-zone.com`.

The use of `bindVars` looks like a great solution.

Reply:
It is true: with `LIKE` clauses, you must escape `_` characters with a preceding `\`, else the `_` is treated as a single-char wildcard.

Reply:
`bindVars` doesn't know when you're using a `LIKE` clause, so doesn't escape the `_`.

Reply:
ok, i now have a better understanding of the `_` in mysql situation.  as to whether a key may exist such as:

`MODULETSHIPPING8STOREPICKUP` that would fit the un-escaped clause is up for debate....  but it would now be addressed, if it did in fact exist.